### ISSUE TITLE
Bug: Incorrect syntax near the keyword 'WHEN'

### ISSUE THREAD

**Issue by AlirezaInGitHub** [zzzprojects/EntityFramework-Extensions#113]
Consider the following entity:
```
public class Role
{
     public Guid Id { get;  set; }
     public string Name { get; set; }
}
```
I was trying to merge roles with the following rules: 
If not found with `Name` then insert it.
If found then no action is required.

So, I setup and executed `BulkMerge` as:
```
var db = new MyDbContext();
var roles = new[]
            {
                new Role { Id = Guid.NewGuid(), Name = "Employees" },
                new Role { Id = Guid.NewGuid(), Name = "Managers" },
                new Role { Id = Guid.NewGuid(), Name = "Administrators" },
            };

db.BulkMerge(roles, options =>
            {
                options.ColumnPrimaryKeyExpression = role => role.Name;
                options.IgnoreOnMergeUpdateExpression = role => role.Id;
            });
```

But I got the following error:

```Incorrect syntax near the keyword 'WHEN'```

Stack trace:
```
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, SqlDataReader ds)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.System.Data.IDbCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.Common.DbDataAdapter.FillInternal(DataSet dataset, DataTable[] datatables, Int32 startRecord, Int32 maxRecords, String srcTable, IDbCommand command, CommandBehavior behavior)
   at System.Data.Common.DbDataAdapter.Fill(DataSet dataSet, Int32 startRecord, Int32 maxRecords, String srcTable, IDbCommand command, CommandBehavior behavior)
   at System.Data.Common.DbDataAdapter.Fill(DataSet dataSet)
   at .(SqlCommand , BulkOperation , Int32 )
   at .( , DbCommand )
   at .(List`1 )
   at Z.BulkOperations.BulkOperation.()
   at .[](DbContext this, IEnumerable`1 , Action`1 )
   at Infrastructure.EntityFramework.DataSeeder.SeedRoles(ConnectDbContext db) in D:\###\EntityFramework\MyDataSeeder.cs:line 219
```
I inspected the generated SQL script:
```
exec sp_executesql N'MERGE INTO [SecurityManagement].[Role] AS DestinationTable
USING
(
SELECT @0_0 AS [Id], @0_1 AS [Name], @0_2 AS ZZZ_Index
UNION ALL SELECT @1_0 AS [Id], @1_1 AS [Name], @1_2 AS ZZZ_Index
UNION ALL SELECT @2_0 AS [Id], @2_1 AS [Name], @2_2 AS ZZZ_Index
) AS StagingTable
ON DestinationTable.[Name] = StagingTable.[Name]
WHEN MATCHED   THEN
    UPDATE
    SET     
WHEN NOT MATCHED THEN
    INSERT ( [Id], [Name] )
    VALUES ( [Id], [Name] )
OUTPUT
    $action,
    StagingTable.ZZZ_Index,
    1,
    INSERTED.[Id] AS [Id_zzzinserted]

;',N'@0_0 uniqueidentifier,@0_1 nvarchar(max) ,@0_2 int,@1_0 uniqueidentifier,@1_1 nvarchar(max) ,@1_2 int,@2_0 uniqueidentifier,@2_1 nvarchar(max) ,@2_2 int',@0_0='5C1DA760-DA70-44AA-AFEB-81F7496D0231',@0_1=N'Employees',@0_2=0,@1_0='02795F86-B0C9-4D00-97DA-CC9A6A583EF7',@1_1=N'Managers',@1_2=1,@2_0='00335E57-C03B-49B9-A48E-F0F5BC1B0781',@2_1=N'Administrators',@2_2=2
```
which is clearly invalid script because `SET` clause is empty.
I think the issue can be explained like this:
The corresponding table has two columns _Id_ and _Name_: Column _Name_ is used for comparison. Column _Id_ is ignored in `SET` clause and because there is no more column to update the `SET` clause ends up empty.

**Comment by JonathanMagnan** [comment]
Hello @AlirezaInGitHub,

Thank you for reporting.

We will look at it and check how to modify our library to handle the case when the UPDATE statement is empty.

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello @AlirezaInGitHub ,

The issue has been fixed in v3.14.26

The `WHEN MATCHED` not longer appears when there is nothing to update.

Best Regards,

Jonathan

**Comment by AlirezaInGitHub** [comment]
Thanks @JonathanMagnan. No issue any more.
